### PR TITLE
[HxMultiSelect] Adds Floating Label type to MultiSelect component

### PR DESCRIPTION
**Pull request by mmonteagudo**
This intends to add Floating Label type to the Multiselect component

### CODE REVIEW

**Comment by mmonteagudo** [comment]
This solves issue https://github.com/havit/Havit.Blazor/issues/900

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor [hunk 1]
```diff
@@ -5,8 +5,7 @@
 @{
 	bool enabled = EnabledEffective && (ItemsToRender != null);
 }
-<div class="@CssClassHelper.Combine("hx-multi-select",
-                                    HasInputGroups ? "input-group" : null,
+<div class="@CssClassHelper.Combine(HasInputGroups ? "input-group" : null,
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor:8]: Why did you move the `.hx-multi-select` class to the inner `div`?
According to our conventions, the "component class" should be applied to the most outer HTML element to allow scoping of any additional CSS rules.

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor [hunk 2]
```diff
@@ -16,7 +15,7 @@
 	}
 
 	@InputGroupStartTemplate
-	<div @ref="_elementReference" class="dropdown" role="listbox">
+    <div @ref="_elementReference" class="hx-multi-select dropdown @(LabelTypeEffective == LabelType.Floating ? "form-floating": "")" role="listbox">
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor:18]: Use `CssClassHelper.Combine()` to build the class list.

File: Havit.Blazor.Components.Web.Bootstrap/Forms/HxInputBase.cs [hunk 3]
```diff
@@ -145,10 +145,27 @@ public abstract class HxInputBase<TValue> : InputBase<TValue>, ICascadeEnabledCo
...
-		((this is IInputWithLabelType inputWithLabelType) && (inputWithLabelType.LabelTypeEffective == LabelType.Floating))
-		? LabelValueRenderOrder.ValueLabel
-		: LabelValueRenderOrder.LabelValue;
+	protected virtual LabelValueRenderOrder RenderOrder
+	{
+		get
+		{
+			if ((this is IInputWithLabelType inputWithLabelType) && (inputWithLabelType.LabelTypeEffective == LabelType.Floating))
+			{
+				if (this.GetType().IsGenericType && this.GetType().GetGenericTypeDefinition() == typeof(HxMultiSelect<,>))
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Forms/HxInputBase.cs:154]: We don't want any such conditions (using reflection and concrete component types) to be present in `HxInputBase`. 
Use `override LabelValueRenderOrder => {put yout component-specific rule here}` in `HxMultiSelect` class instead.

File: Havit.Blazor.Documentation/Pages/Components/HxMultiSelectDoc/HxMultiSelect_Demo_BasicUsage.razor [hunk 4]
```diff
@@ -4,6 +4,7 @@
                TItem="EmployeeDto"
                TValue="int"
                Data="@employees"
+               LabelType="LabelType.Floating"
```
Comment (hakenr) [inline Havit.Blazor.Documentation/Pages/Components/HxMultiSelectDoc/HxMultiSelect_Demo_BasicUsage.razor:7]: Do not modify the basic demo. Please revert.
If you need to test the implementation, use the new `TestApp` or the soon-to-be-removed `BlazorAppTest`

**Review by hakenr — CHANGES_REQUESTED** [review]
@mmonteagudo There are a few points to polish. Please let me know if you want to resolve them or if you prefer us to finish the PR.

@crdo Please check the PR from your POV.

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor (hunk 1 above)
Comment (mmonteagudo) [reply Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor:8]: Thanks it needed to have a correct display of the floating label if the class is not in the inner div it displays incorrect as Date component shows also incorrect when using Floating label (btw, I'm testing a fix for DatePicker component using floating labels)

But maybe we could find a talkaround, that's the one that worked for me, I'm using it in my production environment right now

File: Havit.Blazor.Components.Web.Bootstrap/Forms/HxInputBase.cs (hunk 3 above)
Comment (mmonteagudo) [reply Havit.Blazor.Components.Web.Bootstrap/Forms/HxInputBase.cs:154]: Got it

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor (hunk 1 above)
Comment (hakenr) [reply Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor:8]: Just double-checked and the `hx-multi-select` class isn't used by any existing CSS rules in the library, so moving the class to the inner `div` won't have any effect. We want this class applied to the outermost HTML element to allow scoping for any custom CSS rules the user creates in their project.

File: Havit.Blazor.Documentation/Pages/Components/HxMultiSelectDoc/HxMultiSelect_Demo_BasicUsage.razor (hunk 4 above)
Comment (hakenr) [reply Havit.Blazor.Documentation/Pages/Components/HxMultiSelectDoc/HxMultiSelect_Demo_BasicUsage.razor:7]: Most input components don’t have separate demos for floating labels, as the basic idea is introduced here:  
https://havit.blazor.eu/components/Inputs#floating-labels  
(Which reminds me - we should update the list of inputs supporting floating labels on that page.)  

If you think the `HxMultiSelect` is unique in how it handles floating labels and it’s worth presenting separately, feel free to do so. I’ll leave that decision up to you.

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor (hunk 1 above)
Comment (mmonteagudo) [reply Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor:8]: In my tests it shows the same as the HxInputDate shows look at the attached screenshot:
image

**Review by hakenr — APPROVED** [review]
Thanks again for your contribution! I've consolidated the PR (check out the commits if you're interested), and it's now ready to be merged.

File: Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor (hunk 1 above)
Comment (hakenr) [reply Havit.Blazor.Components.Web.Bootstrap/Forms/Internal/HxMultiSelectInternal.razor]: `HxInputDate` issue extracted to #951
